paymentsheet/src/main/java/com/stripe/android/paymentsheet/DeferredIntentValidator.kt

```diff
@@ -0,0 +1,82 @@
+package com.stripe.android.paymentsheet
+
+import com.stripe.android.model.DeferredIntentParams
+import com.stripe.android.model.ElementsSessionParams
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.PaymentIntent.CaptureMethod
+import com.stripe.android.model.SetupIntent
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.paymentsheet.repositories.toElementsSessionParams
+
+@OptIn(ExperimentalPaymentSheetDecouplingApi::class)
+internal object DeferredIntentValidator {
+
+    fun validate(
+        stripeIntent: StripeIntent,
+        intentConfiguration: PaymentSheet.IntentConfiguration,
+        isFlowController: Boolean,
+    ): StripeIntent {
+        val params = mapToDeferredIntentParams(intentConfiguration)
```

Comment:
Converting `IntentConfiguration` to `DeferredIntentParams` for easier comparison with `StripeIntent`, as the latter two use the same types.